CreeDictionary/tests/CreeDictionary_tests/test_sitemap.py

```diff
@@ -0,0 +1,50 @@
+"""
+Test that the sitemap is accessible and has expected results.
+"""
+
+import random
+import xml.etree.ElementTree as ET
+from http import HTTPStatus
+from urllib.parse import urlparse
+
+import pytest
+from django.urls import reverse
+
+
+@pytest.mark.django_db
+def test_can_access_sitemap(client, sitemap_url):
+    assert sitemap_url == "/sitemap.xml"
+
+    r = client.get(sitemap_url)
+    assert r.status_code == HTTPStatus.OK
+    assert "xml" in r["Content-Type"]
+
+
+@pytest.mark.django_db
+def test_sitemap_has_valid_locations(client, sitemap_url):
+    r = client.get(sitemap_url)
+    root = ET.fromstring(r.content)
+    # Expecting the following:
+    # <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+    #   <!-- primary site urls -->
+    #   <url>
+    #     <loc>http://itwewina.altlab.app/</loc>
+    #   </url>
+    #
+    #   <!-- head word forms -->
+    #   <url>
+    #     <loc>http://itwewina.altlab.app/word/atim/</loc>
+    #   </url>
+    # </urlset>
+    ns = {"sitemap": "http://www.sitemaps.org/schemas/sitemap/0.9"}
+    locations = [loc.text for loc in root.findall("./sitemap:url/sitemap:loc", ns)]
```

Comment:
I'm having a hard time following what's happening here

Reply:
I rewrote it. I always make the same mistake when writing XPath queries with the ElementTree library 😩

Reply:
Thanks! It makes more sense to me now.